lib/internal.js

```diff
@@ -27,15 +27,20 @@ let clientObject;
         return this;
     }
 
-    function sendRequest(userOptions, optionalName,  requestType='single', responseType='single', validStatusCodes = [200, 201, 204, 404]){
+    function sendRequest(userOptions, operationCallback, optionalName, requestType='single', responseType='single'){
```

Comment:
I am thinking the callback should be the second arg, assuming it's likely to populate it. That allows for omitting the last 3 args (if "single" works for both the request and response type).